packages/realm-server/billing/stripe-webhook-handlers/checkout-session-completed.ts

```diff
@@ -0,0 +1,43 @@
+import { DBAdapter } from '@cardstack/runtime-common';
+import {
+  insertStripeEvent,
+  markStripeEventAsProcessed,
+  updateUserStripeCustomerId,
+} from '../billing-queries';
+import { StripeCheckoutSessionCompletedWebhookEvent } from '.';
+
+import PgAdapter from '../../pg-adapter';
+import { TransactionManager } from '../../pg-transaction-manager';
+
+export async function handleCheckoutSessionCompleted(
+  dbAdapter: DBAdapter,
+  event: StripeCheckoutSessionCompletedWebhookEvent,
+): Promise<Response> {
+  let txManager = new TransactionManager(dbAdapter as PgAdapter);
+
+  await txManager.withTransaction(async () => {
+    try {
+      await insertStripeEvent(dbAdapter, event);
+    } catch (error) {
+      if (
+        error instanceof Error &&
+        error.message.includes('duplicate key value')
+      ) {
+        throw new Error('Stripe event already processed');
+      }
+      throw error;
+    }
+
+    const stripeCustomerId = event.data.object.customer;
+    const matrixUserName = event.data.object.client_reference_id;
+
+    await updateUserStripeCustomerId(
+      dbAdapter,
+      matrixUserName,
+      stripeCustomerId,
+    );
+    await markStripeEventAsProcessed(dbAdapter, event.id);
+  });
+
+  return new Response('ok');
```

Comment:
Is this needed?

Reply:
Not actually, I have removed it - thanks